File: core/src/main/java/net/minestom/server/entity/vehicle/PlayerVehicleInformation.java
package net.minestom.server.entity.vehicle;

public class PlayerVehicleInformation {

    private float sideways;
    private float forward;
    private boolean jump;
    private boolean unmount;

    public float getSideways() {
        return sideways;
    }

    public float getForward() {
        return forward;
    }

    public boolean shouldJump() {
        return jump;
    }

    public boolean shouldUnmount() {
        return unmount;
    }

    /**
     * Refresh internal data
     *
     * @param sideways the new sideways value
     * @param forward  the new forward value
     * @param jump     the new jump value
     * @param unmount  the new unmount value
     */
    public void refresh(float sideways, float forward, boolean jump, boolean unmount) {
        this.sideways = sideways;
        this.forward = forward;
        this.jump = jump;
        this.unmount = unmount;
    }
}

File: src/main/java/net/minestom/server/entity/vehicle/PlayerVehicleInformation.kt
package net.minestom.server.entity.vehicle

class PlayerVehicleInformation {
    var sideways = 0f
        private set
    var forward = 0f
        private set
    private var jump = false
    private var unmount = false
    fun shouldJump(): Boolean {
        return jump
    }

    fun shouldUnmount(): Boolean {
        return unmount
    }

    /**
     * Refresh internal data
     *
     * @param sideways the new sideways value
     * @param forward  the new forward value
     * @param jump     the new jump value
     * @param unmount  the new unmount value
     */
    fun refresh(sideways: Float, forward: Float, jump: Boolean, unmount: Boolean) {
        this.sideways = sideways
        this.forward = forward
        this.jump = jump
        this.unmount = unmount
    }
}